### PR TITLE
Scala Native 0.4.0-M2

### PR DESCRIPTION
**Pull request by cheeseng**
Updated to use Scala Native 0.4.0-M2.

### CODE REVIEW

File: project/scalatest.scala [hunk 1]
```diff
@@ -1152,7 +1153,7 @@ object ScalatestBuild {
       initialCommands in console := """|import org.scalatest._
                                        |import org.scalactic._
                                        |import Matchers._""".stripMargin,
-      libraryDependencies += "org.scala-native" %%% "test-interface" % "0.3.6",
+      libraryDependencies += "org.scala-native" %% "test-interface_native0.4.0-M2" % "0.4.0-M2",
```
Comment (lolgab) [inline project/scalatest.scala]: You could use:
```scala
libraryDependencies += "org.scala-native" %%% "test-interface" % nativeVersion
```

File: project/scalatest.scala [hunk 2]
```diff
@@ -16,9 +16,10 @@ import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport.{scalaJSLinkerConfig, jsEn
 import sbtcrossproject.CrossPlugin.autoImport._
 
 import scalanative.sbtplugin.ScalaNativePlugin
-import scalanative.tools
-import scalanative.optimizer.{inject, pass}
-import scalanative.sbtplugin.ScalaNativePluginInternal.{nativeConfig, nativeOptimizerDriver, nativeLinkerReporter, nativeOptimizerReporter, NativeTest}
+//import scalanative.tools
```
Comment (lolgab) [inline project/scalatest.scala]: Are you leaving the commented parts? I don't think they will be useful in the future..

File: project/scalatest.scala [hunk 3]
```diff
@@ -1448,7 +1449,7 @@ object ScalatestBuild {
         organization := "org.scalatest",
         moduleName := "scalatest-app",
         libraryDependencies ++= nativeCrossBuildLibraryDependencies.value,
-        libraryDependencies += "org.scala-native" %%% "test-interface" % "0.3.6",
+        libraryDependencies += "org.scala-native" %% "test-interface_native0.4.0-M2" % "0.4.0-M2",
```
Comment (lolgab) [inline project/scalatest.scala]: Same here:
```scala
libraryDependencies += "org.scala-native" %%% "test-interface" % nativeVersion
```

**Comment by bvenners** [comment]
@cheeseng I forgot to mention that this did not build for me. Will connect with you on it.

File: project/scalatest.scala (hunk 3 above)
Comment (cheeseng) [reply project/scalatest.scala]: @lolgab Adjustments made, thanks for pointing them out.

**Comment by bvenners** [comment]
Figured out the build problem was my mistake. Merging now.

**Pull request merged**